src/bin/sol-fbp-generator/main.c

```diff
@@ -633,8 +658,12 @@ generate_node_type_assignments(const struct fbp_data *data)
     out("\n");
 
     SOL_VECTOR_FOREACH_IDX (&data->graph.nodes, n, i) {
-        struct type_description *desc = n->user_data;
-        out("    nodes[%d].type = %s;\n", i, desc->symbol);
+        struct node_data *nd = n->user_data;
+
+        if (nd->is_declared)
```

Comment:
hum... maybe check `nd->desc` or `nd->desc->symbol` and remove `nd->is_declared`? is it possible?